core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedScheduleGenerator.scala

```diff
@@ -606,8 +610,8 @@ class CostBasedScheduleGenerator(
       .map(level =>
         level
           .map(region => {
-            val (resourceConfig, regionCost) =
-              costEstimator.allocateResourcesAndEstimateCost(region, 1)
+            val (resourceConfig, regionCost) = costEstimatorMemoization
+              .getOrElseUpdate(region, costEstimator.allocateResourcesAndEstimateCost(region, 1))
```

Comment:
Do we really need the entire region as the key? That's a lot of information.

Reply:
We don't, but we do need all the Ids in the region (except regionId) to uniquely identify a region so that it contains all the information needed by the costEstimator and resourceAllocator. I created a separate case class for this key.